crits/samples/api.py

```diff
@@ -113,21 +113,39 @@ def obj_create(self, bundle, **kwargs):
 
         result = {'success': False}
 
-        if len(sample_md5) > 0:
-            result = sample_md5[0]
-            if result.get('message'):
-                content['message'] = result.get('message')
-            if result.get('object'):
-                content['id'] = str(result.get('object').id)
-            if content.get('id'):
-                url = reverse('api_dispatch_detail',
-                            kwargs={'resource_name': 'samples',
-                                    'api_name': 'v1',
-                                    'pk': content.get('id')})
-                content['url'] = url
+        if len(sample_results) > 0:
+            result = sample_results[0]
+
+            if type(result) == dict:
+                # Metadata for raw file
+                if result.get('message'):
+                    content['message'] = result.get('message')
+                if result.get('object'):
+                    content['id'] = str(result.get('object').id)
+
+                    page_url = reverse('crits.samples.views.detail',
+                                args=[str(result.get('object').md5)])
+                    content['page_url'] = page_url
+                if content.get('id'):
+                    api_url = reverse('api_dispatch_detail',
+                                kwargs={'resource_name': 'samples',
+                                        'api_name': 'v1',
+                                        'pk': content.get('id')})
+
+                    content['url'] = api_url
+
+                if result.get('success'):
+                    content['return_code'] = 0
+
+            else:
+                # Metadata for Zip, RAR, etc
+                content['id'] = result
+                page_url = reverse('crits.samples.views.detail',
+                                   args=[result])
+
+                content['page_url'] = page_url
```

Comment:
because `url` is an API endpoint we can hit to get data about the sample but we have an application that will need to somehow get a direct link to the sample's Crits page or the download link for the sample itself.